src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java

```diff
@@ -247,14 +246,18 @@ private void sendEmailForPayment(final Account account, final ExtBusEvent killbi
             }
         }
 
-        if (invoice.getNumberOfPayments() == 0) {
-            logService.log(LogService.LOG_INFO, String.format("Zero payment invoice found for invoice: %s", invoiceId.toString()));
-            if (invoice.getBalance().compareTo(BigDecimal.ZERO) == 0 && recurringPayment && !oneTimePayment) {
+        if (killbillEvent.getEventType() == ExtBusEventType.INVOICE_CREATION) {
+            // We only want to send requests for INVOICE_CREATION events that are not also generating INVOICE_PAYMENT events
+            // i.e. those with no payments, and a zero balance. All others are ignored as they were before.
+            // This work only effects invoices with ONLY recurring items.
+            if (invoice.getNumberOfPayments() == 0) {
+                if (invoice.getBalance().compareTo(BigDecimal.ZERO) == 0 && recurringPayment && !oneTimePayment) {
+                    logService.log(LogService.LOG_INFO, String.format("Generating request for invoice: %s", invoiceId.toString()));
                     subscriptionClient.sendEmailRequest("Purchase_Success", invoice, account);
-            } else {
-                logService.log(LogService.LOG_INFO, String.format("Invoice: %s failed check with BalComp: %i, recurring: %b, and oneTime: %b",
-                        invoiceId.toString(), invoice.getBalance().compareTo(BigDecimal.ZERO), recurringPayment, oneTimePayment));
+                }
             }
+            logService.log(LogService.LOG_INFO, String.format("Invoice: %s failed check with # of payments: %i, BalComp: %i, recurring: %b, and oneTime: %b",
+                invoiceId.toString(), invoice.getNumberOfPayments(), invoice.getBalance().compareTo(BigDecimal.ZERO), recurringPayment, oneTimePayment));
```

Comment:
why log this on a "successful" 0 payment invoice? Or throw a return statement right after calling subscription service on the success path.

Reply:
You're pretty sharp, I'm going to guess you've done this before